### DISCUSSION TITLE
Whole Preview Section is not updating when changing branches on repository homepage

### DISCUSSION THREAD

**Discussion by zhravan** [community/community#101898] [Repositories]
### Select Topic Area

Bug

## Description:

Upon switching branches on the repository homepage, the README content does not dynamically update to reflect the contents of the currently selected branch, persisting with the content of the default main branch until the README file is explicitly clicked. This behavior appears to have surfaced after the latest UI release changes.

## Steps to Reproduce:

- Navigate to the repository homepage on GitHub, starting on the default main branch where the latest README contents are displayed.

- Create and push changes in a new branch (e.g., feat branch) that modifies the README.

- Return to the repository homepage, which defaults to displaying the main branch.

- Change the branch to the previously created feat branch.

- Despite switching branches, the README content displayed on the homepage remains that of the main branch until explicitly clicking on the README file.

## Expected Behavior:

Upon changing the branch on the repository homepage, the README content should dynamically update to reflect the contents of the currently selected branch.

## Additional Information:

This issue seems to have emerged after the latest UI release changes. Please verify if this behavior aligns with the intended design or if it is indeed a bug.

**Comment by RobertNeat** [comment]
The same problem appeared for me - and I hope it's not the test change for the platform (i'm trying to compact my repositories by having it in different branches so i need readme for each branch)

**Reply by zhravan** [reply]
Yes, this is ideally happening only at preview level. Internally seems to be fine.

**Comment by alinkedd** [comment]
The issue extends beyond just the readme file; it affects the entire preview section, as it replaces all the preview content with that from the default branch.

For instance, here https://github.com/github/markup/tree/v0.1.0 we can observe a Code of conduct which is not present in that specific version.

Additionally, there are some peculiar behaviors:
- Initially, the correct content is displayed, but it is subsequently replaced by the content from the main branch.
- If the default branch lacks a readme file, the readme from another branch is loaded and then replaced with an 'Add a README' fallback message.
- If a branch has no root-level files other than the readme (though folders are permitted), the correct readme is displayed.

Overall, it appears that an observable or a similar structure got broken.

**Comment by djsudduth** [comment]
Same problem just appeared for me yesterday. My branch README preview will not update no matter how many commits and pushes to it.

**Comment by zhravan** [comment]
Is there any way we can bump or convey this to the team GitHub??

**Comment by sickozell** [comment]
Same problem for me.
I reported this issue, referencing this thread, here:
https://support.github.com/contact/bug-report

**Reply by bassamanator** [reply]
I just did the same. Thanks.

**Comment by alinkedd** [comment]
Same/duplicate discussion is here https://github.com/orgs/community/discussions/101835

**Comment by sickozell** [comment]
The bug seems fixed now

**Reply by bassamanator** [reply]
Yeah my repo seems fine now, hopefully it stays like that.
